### PR TITLE
update ConstraintStrategy typing

### PR DESCRIPTION
**Pull request by matthyk**
In #179 it was asked whether it is possible to use objects as values in custom constraints. After looking through the code, this should not be a problem. Only the typings have to be updated.
The default value of the generic could be `string` or `any`. 

`any` would match the typing of the RouteOption. 

```javascript
interface RouteOptions {
    constraints?: { [key: string]: any }
  }
```

But `string` would provide a backwards compatibility.

### CODE REVIEW

**Review by mcollina — CHANGES_REQUESTED** [review]
Please update the tests as well.

**Comment by matthyk** [comment]
Should I only update the tsd tests or also add a test to test custom constraints with objects as values?

**Comment by matthyk** [comment]
Which tests exactly need to be updated? For me it looks like it would make most sense to add a new tests for custom constraints with objects. ( like `constraint.custom-with-objects.test` )

**Comment by mcollina** [comment]
Adding new tests is totally fine

**Comment by mcollina** [comment]
We just need to test the types: https://github.com/delvedor/find-my-way/blob/master/test/types/router.test-d.ts

**Review by mcollina — APPROVED** [review]
lgtm

File: index.d.ts [hunk 1]
```diff
@@ -57,17 +57,17 @@ declare namespace Router {
...
 
-  interface ConstraintStrategy<V extends HTTPVersion> {
+  interface ConstraintStrategy<V extends HTTPVersion, T = string> {
     name: string,
     mustMatchWhenDerived?: boolean,
     storage() : {
-      get(version: String) : Handler<V> | null,
-      set(version: String, store: Handler<V>) : void,
-      del(version: String) : void,
+      get(version: T) : Handler<V> | null,
```
Comment (airhorns) [inline index.d.ts:64]: Would you mind updating this variable to be named `value` or something less version specific? Not strictly a result of your change but it is a bit more confusing now that the type of this piece is generic

**Review by airhorns — APPROVED** [review]
LGTM, nice change!

File: index.d.ts (hunk 1 above)
Comment (matthyk) [reply index.d.ts:64]: Perhaps the name of the `store` variable should also be changed, as this has simply been taken over from the semver-store?

Comment (airhorns) [reply index.d.ts:64]: Oh -- yep definitely, that's a `handler` not a `store`. Thanks for leaving it better than you found it!!

**Review by delvedor — APPROVED** [review]
LGTM

**Pull request merged**